### PR TITLE
Handle multiple matching issuers in OCSP requests

### PR DESCRIPTION
**Pull request by stevendpclark**
Select the first issuer that matches our request hashes and has the OCSP signing usage enabled. This might not match the exact issuer id that issued the certificate but the signatures will be okay.

This should resolve the comment brought up by @kitography in [PR#16723](https://github.com/hashicorp/vault/pull/16723#discussion_r951592384). Thanks again for identifying this issue in the review!

### CODE REVIEW

File: builtin/logical/pki/ocsp_test.go [hunk 1]
```diff
@@ -309,6 +309,63 @@ func TestOcsp_RevokedCertHasIssuerWithoutAKey(t *testing.T) {
...
+	requireSuccessNonNilResponse(t, resp, err, "failed resetting usage flags on issuer")
+	requireFieldsSetInResp(t, resp, "usage")
+	// Do not assume a specific ordering for usage...
+	usages, err := NewIssuerUsageFromNames(strings.Split(resp.Data["usage"].(string), ","))
+	require.NoError(t, err, "failed parsing usage return value")
+	require.True(t, usages.HasUsage(IssuanceUsage))
+	require.True(t, usages.HasUsage(CRLSigningUsage))
+	require.False(t, usages.HasUsage(OCSPSigningUsage))
+
+	// Request an OCSP request from it, we should get an Unauthorized response back
```
Comment (kitography) [inline builtin/logical/pki/ocsp_test.go:346]: This comment is out of place, and not what this test is checking?

File: builtin/logical/pki/ocsp_test.go [hunk 2]
```diff
@@ -309,6 +309,63 @@ func TestOcsp_RevokedCertHasIssuerWithoutAKey(t *testing.T) {
...
+	respDer := resp.Data["http_raw_body"].([]byte)
+
+	ocspResp, err := ocsp.ParseResponse(respDer, testEnv.issuer1)
+	require.NoError(t, err, "parsing ocsp get response")
+
+	require.Equal(t, ocsp.Good, ocspResp.Status)
+	require.Equal(t, crypto.SHA1, ocspResp.IssuerHash)
+	require.Equal(t, 0, ocspResp.RevocationReason)
+	require.Equal(t, testEnv.leafCertIssuer1.SerialNumber, ocspResp.SerialNumber)
+	require.Equal(t, crypto.SHA1, ocspResp.IssuerHash)
```
Comment (kitography) [inline builtin/logical/pki/ocsp_test.go:361]: nit: same as line 358?

File: builtin/logical/pki/ocsp_test.go [hunk 3]
```diff
@@ -309,6 +309,63 @@ func TestOcsp_RevokedCertHasIssuerWithoutAKey(t *testing.T) {
...
+	require.Equal(t, rotatedCert, ocspResp.Certificate)
+	require.Equal(t, 0, ocspResp.RevocationReason)
```
Comment (kitography) [inline builtin/logical/pki/ocsp_test.go:363]: nit: same check as 359?

**Review by kitography — APPROVED** [review]
This code looks excellent, I have some nits on the later half of the test

File: builtin/logical/pki/ocsp_test.go (hunk 1 above)
Comment (stevendpclark) [reply builtin/logical/pki/ocsp_test.go]: Yup sadly copy/paste let me down, going a little too quickly. Thanks for feedback, the other items have been addressed in the feedback commit. Much appreciated!

**Review by cipherboy — APPROVED** [review]

**Pull request merged**